cluster/gce/windows/k8s-node-setup.psm1

```diff
@@ -1113,13 +1113,26 @@ $STACKDRIVER_ROOT = 'C:\Program Files (x86)\Stackdriver'
 # sometimes is unstoppable, so we work around it by killing the processes.
 function Restart-StackdriverLoggingAgent {
   Stop-Service -NoWait -ErrorAction Ignore StackdriverLogging
-  # TODO: check periodically to lower the wait time
-  Start-Sleep 10
+ 
+  # Wait (if necessary) for service to stop.
+  $timeout = 10
+  $stopped = (Get-service StackdriverLogging).Status -eq 'Stopped'
+  for ($i = 0; $i -lt $timeout -and !($stopped); $i++) {
+      Start-Sleep 1
+      $stopped = (Get-service StackdriverLogging).Status -eq 'Stopped'
+  }
+
   if ((Get-service StackdriverLogging).Status -ne 'Stopped') {
     # Force kill the processes.
     Stop-Process -Force -PassThru -Id (Get-WmiObject win32_process |
       Where CommandLine -Like '*Stackdriver/logging*').ProcessId
   }
+
+  # Wait until process has stopped.
+  while ((Get-service StackdriverLogging).Status -ne 'Stopped') {
+      Start-Sleep 1
```

Comment:
This can wait indefinitely. Could you add some periodic logging (e.g., maybe every 10s) so that we know what's going on?

Reply:
The periodic logging looks good, please also raise an exception if the process still hasn't stopped after some timeout (60 seconds?). That would be preferable to waiting indefinitely.

See `Throw` examples already in this file, and https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_throw?view=powershell-6 for full details.